app/controllers/queue_items_controller.rb

```diff
@@ -31,4 +42,17 @@ def item_is_already_queued?(video)
   def remove_item_from_queue(queue_item)
     queue_item.destroy if current_user.queue_items.include?(queue_item)
   end
+
+  def update_item_array(items_array)
+      ActiveRecord::Base.transaction do
+        items_array.each do |item|
+          update_item(item)
+        end
+      end
+  end
+
+  def update_item(item)
+    queue_item = QueueItem.find(item["id"])
+    queue_item.update_attributes!(position: item["position"], rating: item["rating"]) if queue_item.user == current_user

```

Comment:
In Rails 4, you can use just `.update!` instead of `.update_attributes!` to do the same.